DotNet/DataAcquisition/Jobs/AcquisitionProcessingJob.cs

```diff
@@ -230,33 +230,75 @@ public async Task ProcessPendingTailingMessages(CancellationToken cancellationTo
 
         try
         {
-            var originalParentId = Activity.Current?.ParentId;
-
             foreach (var message in tailingMessages)
             {
                 try
                 {
-                    using var activity = ServiceActivitySource.Instance.StartActivity("AcquisitionProcessingJob.ProcessPendingTailingMessages");
-
-                    _logger.LogDebug("Setting tail message parent id to {TraceParentId}", message.TraceParentId ?? "null");
-                    activity.SetParentId(message.TraceParentId ?? originalParentId);
-                    activity.AddTag("link.correlation_id", message.CorrelationId);
-                    activity.AddTag("link.facility_id", message.FacilityId);
-                    activity.AddTag("link.report_tracking_id",
-                        message.ResourceAcquired.ScheduledReports.FirstOrDefault()?.ReportTrackingId ?? string.Empty);
-
-                    activity.Start();
-
+                    // Parse the traceparent string (format: 00-traceId-spanId-flags)
```

Comment:
Not essential... But it might be good to extract this logic into a separate method to separate "trace" logic from other business logic.